rewrite-core/src/main/java/org/openrewrite/shell/exec/ShellExecutor.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.shell.exec;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.scheduling.WorkingDirectoryExecutionContextView;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@org.openrewrite.Incubating(since = "8.30.0")
+public interface ShellExecutor {
+
+    @SuppressWarnings("unused")
+    default void init() {}
```

Comment:
Did you want to add the `Path dir` here and below in `postExec()`that you had looked to want to override?